tsdb/engine/tsm1/tombstone.go

```diff
@@ -183,43 +182,51 @@ func (t *Tombstoner) Delete() error {
 
 // HasTombstones return true if there are any tombstone entries recorded.
 func (t *Tombstoner) HasTombstones() bool {
-	files := t.TombstoneFiles()
+	stats := t.TombstoneStats()
+	if !stats.TombstoneExists {
+		return false
+	}
+	if stats.Size > 0 {
+		return true
+	}
```

Comment:
Not sure if it's worth the verbosity to avoid the extra lock below

Reply:
I think it's good as long as a non-zero size always means there are tombstones.

Reply:
It's the same end behavior as before, at least